Automattic/woocommerce-payments
# Fix payment failure when phone number is hidden

Fixes #3029

#### Changes proposed in this Pull Request
When the phone number field is disabled, making a payment via Checkout Block with UPE enabled makes the payment fail. This is similar to the issue in classic UPE payments fixed with https://github.com/Automattic/woocommerce-payments/pull/2853. Changes proposed here follow the same workaround of passing in a truthy value when fields are empty. 

#### Additional notes

I experimented with removing the `phone` field from the object we pass to Stripe (instead of setting it to a truthy value) but that results in the following error

![Screen Shot 2021-10-06 at 1 53 50 PM](https://user-images.githubusercontent.com/6216000/136271119-fc9af15b-5e4d-4d77-85dd-4fb6a7dd3a18.png)

I could not find any documentation on `elementOptions` but if we don't include hidden fields in [there](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/blocks/upe-fields.js#L179), then we can also exclude them here in `confirmParams`

#### e2e tests
Will be covered under https://github.com/Automattic/woocommerce-payments/issues/1610








#### Testing instructions





1. Create a checkout page using the [checkout block](https://wordpress.org/plugins/woo-gutenberg-products-block/).
2. In the block options, disable the phone field.
3. Enable UPE checkout in WCPay Dev utils
4. Checkout a product using the newly created checkout page.
5. Notice that payment completes successfully without an error.

-------------------


**Post merge**
